erpnext/selling/utils/order_for.py

```diff
@@ -39,7 +39,7 @@ def initialise_order_for():
 			order_for["customer_primary_contact_name"] = contact.name
 
 	# Keep customer info up to date on every session start
-	customer = get_party()
+	customer = get_party(party_doctype="Customer", party=order_for.get("customer_name"))
```

Comment:
why this change?

Reply:
It tries to create a new customer once if you look at cart.py. you will understand this

Reply:
This makes sense, it enforces a customer type when using order-for functionality.